tests/core/test_state_store.py

```diff
@@ -64,6 +66,7 @@ def test_is_checkpoint_updated():
     assert s2.is_checkpoint_updated()
 
     s2.update_state("test-state2")
+    time.sleep(1)
```

Comment:
why is this needed?

Reply:
same as above

Reply:
As I said earlier, I believe you shouldn't need this and below